promise.js

```diff
@@ -119,15 +119,47 @@ class PromiseConnection extends EventEmitter {
     });
   }
 
-  beginTransaction() {
+  beginTransaction(options) {
     const c = this.connection;
     const localErr = new Error();
     return new this.Promise((resolve, reject) => {
       const done = makeDoneCb(resolve, reject, localErr);
-      c.beginTransaction(done);
+      c.beginTransaction(options,done);
     });
   }
 
+  transaction(options,promiseCallback) {
+    if (! promiseCallback) {
+      promiseCallback = options;
+      options = {};
+    }
+    options = options || {};
+  
+    let promiseChain = this.Promise.resolve();
+
+    if (options.autoCommit !== true) {
+      promiseChain = promiseChain.then(() => this.beginTransaction(options));
+    }
+  
+    promiseChain = promiseChain.then(() => promiseCallback(this));
+
+    if (options.autoCommit !== true) {
+      promiseChain = promiseChain
+      .then( res => 
+        this.commit()
+        .then(
+          () => res
+        )
+      )
+      .catch( err => 
+        this.rollback()
+        .catch(() => null)
```

Comment:
This should probably emit an error or even override the previous error, it certainly shouldn't be swallowed.